lib/galaxy/config.py

```diff
@@ -276,6 +276,11 @@ def __init__( self, **kwargs ):
         # On can mildly speed up Galaxy startup time by disabling index of help,
         # not needed on production systems but useful if running many functional tests.
         self.index_tool_help = string_as_bool( kwargs.get( "index_tool_help", True ) )
+        self.tool_name_boost = kwargs.get( "tool_name_boost", None )
```

Comment:
You had argued against additional complexity without reason.  I would suggest that this implementation IS that additional complexity without any reason at all.

Everything else in galaxy that uses config this way, and I think a strong argument needs to be made for changing this practice.